bad/source/lex.rs

```diff
@@ -0,0 +1,87 @@
+use logos::Logos;
+
+#[derive(Logos, Debug)]
+pub enum Token {
+	#[regex(r"[ \t\n\f]")]
+	Whitespace,
+	
+	#[regex(r"[a-zA-Z_]+[a-zA-Z0-9_]*")]
+	Identifier,
+	Symbol,
+	Number,
+	Comment,
+	// Keywords
+	#[token("if")]

```

Comment:
There's an interesting philosophical question of whether the lexer should have a single token for all words (and have the parser decide if a word is a keyword or not). I've done both, and over time, I'm not sure if I love one over the other that much...

Reply:
identifer with a bit for "is keyword" versus "every keyword is its own special token".... I don't know which I'd prefer either, tbh...